realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -462,31 +458,19 @@ public TableView findAll() {
     // handover find* methods
     // this will use a background SharedGroup to import the query (using the handover object)
     // run the query, and return the table view to the caller SharedGroup using the handover object.
-    public long findAllWithHandover(long bgSharedGroupPtr, long nativeReplicationPtr,  long ptrQuery) throws BadVersionException {
-        validateQuery();
```

Comment:
The problem this method is calling in a worker thread and validating the query created in the caller thread, which is quite strange.